sql/core/src/test/scala/org/apache/spark/sql/GeneratorFunctionSuite.scala

```diff
@@ -89,4 +91,30 @@ class GeneratorFunctionSuite extends QueryTest with SharedSQLContext {
       exploded.join(exploded, exploded("i") === exploded("i")).agg(count("*")),
       Row(3) :: Nil)
   }
+
+  test("inline with empty table or empty array") {
```

Comment:
the test name is misleading: we do allow empty array, the problem is `array()` returns an array of null, which fails the type check.